public/app/plugins/panel/barchart/bars.ts

```diff
@@ -108,53 +194,83 @@ export function getConfig(opts: BarsOptions) {
     );
   };
 
-  const drawPoints: Series.Points.Show =
-    formatValue == null
-      ? false
-      : (u, sidx) => {
-          u.ctx.font = font;
-          u.ctx.fillStyle = 'white';
+  // uPlot hook to draw the labels on the bar chart
+  const draw = (u: uPlot) => {
```

Comment:
hmm, i guess this needs to be in this later hook so we can capture label sizes across all series, since `series.paths` and `points.show` fire in alternating sequence. using a `draw` hook is a bit brute force, but probably the simplest changeset for this PR :+1: .

i'll see if i can come up with something a bit more elegant in a follow-up.

Reply:
what's brute force about it?